python/ipywidgets/ipywidgets/__init__.py

```diff
@@ -21,28 +21,36 @@
 from ._version import __version__, __protocol_version__, __jupyter_widgets_controls_version__, __jupyter_widgets_base_version__
 
 import os
+
+from traitlets import link, dlink
 from IPython import get_ipython
+try:
+    from comm import get_comm_manager
+except ImportError:
+    def get_comm_manager():
+        return get_ipython().kernel.comm_manager
+
 from .widgets import *
-from traitlets import link, dlink
+
 
 def load_ipython_extension(ip):
     """Set up Jupyter to work with widgets"""
     if not hasattr(ip, 'kernel'):
         return
-    register_comm_target(ip.kernel)
+    register_comm_target()
 
-def register_comm_target(kernel=None):
+def register_comm_target():
```

Comment:
```suggestion
def register_comm_target(kernel=None):
```
I think we shouldn't change the API, as that would be a breaking change.